src/power_grid_model_io/converters/pandapower_converter.py

```diff
@@ -0,0 +1,507 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model IO project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+"""
+Panda Power Converter
+"""
+import re
+from functools import lru_cache
+from typing import Dict, Mapping, Optional, Union
+
+import numpy as np
+import pandas as pd
+from power_grid_model import Branch3Side, BranchSide, LoadGenType, initialize_array
+from power_grid_model.data_types import Dataset
+
+from power_grid_model_io.converters.base_converter import BaseConverter
+from power_grid_model_io.data_types import ExtraInfoLookup
+from power_grid_model_io.filters import get_winding
+
+StdTypes = Mapping[str, Mapping[str, Mapping[str, Union[float, int, str]]]]
+PandasData = Mapping[str, pd.DataFrame]
+
+CONNECTION_PATTERN_PP = re.compile(r"(Y|YN|D|Z|ZN)(y|yn|d|z|zn)\d*")
+CONNECTION_PATTERN_PP_3WDG = re.compile(r"(Y|YN|D|Z|ZN)(y|yn|d|z|zn)(y|yn|d|z|zn)\d*")
+
+
+class PandaPowerConverter(BaseConverter[PandasData]):
+    """
+    Panda Power Converter
+    """
+
+    __slots__ = ("std_types", "pp_data", "pgm_data", "idx", "idx_lookup", "next_idx", "system_frequency")
+
+    def __init__(self, std_types: Optional[StdTypes] = None, system_frequency: float = 50.0):
+        super().__init__(source=None, destination=None)
+        self.std_types: StdTypes = std_types if std_types is not None else {}
+        self.system_frequency: float = system_frequency
+        self.pp_data: PandasData = {}
+        self.pgm_data: Dataset = {}
+        self.idx: Dict[str, pd.Series] = {}
+        self.idx_lookup: Dict[str, pd.Series] = {}
+        self.next_idx = 0
+
+    def _parse_data(self, data: PandasData, data_type: str, extra_info: Optional[ExtraInfoLookup] = None) -> Dataset:
+
+        # Clear pgm data
+        self.pgm_data = {}
+        self.idx_lookup = {}
+        self.next_idx = 0
+
+        # Set pandas data
+        self.pp_data = data
+        # Convert
+        if data_type == "input":
+            self._create_input_data()
+        else:
+            raise NotImplementedError()
```

Comment:
Do we want this to be more specific? e.g. something like:
`raise NotImplementedError(f"Data type: '{data_type}' is not implemented")`

Reply:
Yes, that would be better. Or maybe we should raise a `ValueError` in stead. I don't think we will implement other data types in the future.

Reply:
Yes, I think `ValueError` would be better indeed